libs/pika/command_line_handling/include/pika/command_line_handling/get_env_var_as.hpp

```diff
@@ -0,0 +1,19 @@
+//  Copyright (c) 2023 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <pika/config/export_definitions.hpp>
+
+#include <cstdlib>
+#include <string>
+
+namespace pika::detail {
+
+    /// from env var name 's' get value if well-formed, otherwise return default
+    template <typename T>
+    PIKA_EXPORT T get_env_var_as(const char* s, T def);
```

Comment:
Is it worth it to hide the definition of this? You could keep the definition in the header but add `extern` declarations for the 7 most common types that you've explicitly instantiated.

Reply:
normally, I would have just put the code in the header template, but in hpx, hartmut removed all my print stuff to improve compilation times, so I put everything in the cpp file for the same reason. I'll happily just put the code directly in the headeer and there's not need for any extern declarations ...